openwisp_controller/geo/admin.py

```diff
@@ -81,5 +81,22 @@ class DeviceLocationInline(ObjectLocationMixin, admin.StackedInline):
 admin.site.register(Location, LocationAdmin)
 
 
+class DeviceLocationFilter(admin.SimpleListFilter):
+    title = _('has geographic position set?')
+    parameter_name = 'with_geo'
+
+    def lookups(self, request, model_admin):
+        return (
+            ('true', _('Yes')),
+            ('false', _('No')),
+        )
+
+    def queryset(self, request, queryset):
+        if self.value():
+            return queryset.filter(devicelocation__isnull=self.value() == 'false')
+        return queryset
```

Comment:
This is missing a test.
I also need to double check if it respects multi-tenancy.